base_delivery_carrier_label/stock.py

```diff
@@ -0,0 +1,307 @@
+                label_obj.with_context(context_attachment).create(data)
+        return True
+
+    @api.multi
+    def action_generate_carrier_label(self):
+        """ Method for the 'Generate Label' button.
+
+        It will generate the labels for all the packages of the picking.
+
+        """
+        return self.generate_labels()
+
+    @api.onchange('carrier_id')
+    def carrier_id_change(self):
+        """ Inherit this method in your module """
+        if not self.carrier_id:
+            return
+        # This can look useless as the field carrier_code and
+        # carrier_type are related field. But it's needed to fill
+        # this field for using this fields in the view. Indeed the
+        # module that depend of delivery base can hide some field
+        # depending of the type or the code
+        carrier = self.carrier_id
+        self.carrier_type = carrier.type
+        self.carrier_code = carrier.code
+        self.option_ids = carrier.default_options()
+        result = {
+            'domain': {
+                'option_ids': [('id', 'in', carrier.available_option_ids.ids)],
+            }
+        }
+        return result
+
+    @api.onchange('option_ids')
+    def option_ids_change(self):
+        if not self.carrier_id:
+            return
+        carrier = self.carrier_id
+        for available_option in carrier.available_option_ids:
+            if (available_option.mandatory
+                    and available_option not in self.option_ids):
+                # XXX the client does not allow to modify the field that
+                # triggered the onchange:
+                # https://github.com/odoo/odoo/issues/2693#issuecomment-56825399
+                # Ideally we should add the missing option
+                raise exceptions.Warning(
+                    _("You should not remove a mandatory option."
+                      "Please cancel the edit or "
+                      "add back the option: %s.") % available_option.name
+                )
+
+    @api.model
+    def _values_with_carrier_options(self, values):
+        values = values.copy()
+        carrier_id = values.get('carrier_id')
+        option_ids = values.get('option_ids')
+        if carrier_id and not option_ids:
+            carrier_obj = self.env['delivery.carrier']
+            carrier = carrier_obj.browse(carrier_id)
+            default_options = carrier.default_options()
+            if default_options:
+                values.update(option_ids=[(6, 0, default_options.ids)])
+        return values
+
+    @api.multi
+    @api.returns('stock.quant.package')
+    def _get_packages_from_picking(self):
+        """ Get all the packages from the picking """
+        self.ensure_one()
+        operation_obj = self.env['stock.pack.operation']
+        operations = operation_obj.search(
+            [('result_package_id', '!=', False),
+             ('picking_id', '=', self.id)]
+        )
```

Comment:
> The packages have no result_package_id, so no labels are generated.

If I guess (can't say "understand" at this point) correctly, the `stock.pack.operation` could have a `package_id`, a `result_package_id` or both. If we have only a `package_id`, we should keep this package. So, the logic should be changed throughout the module to look first in `result_package_id` then in `package_id`.
Does this analysis sounds good to you?